django/db/migrations/state.py

```diff
@@ -282,7 +278,6 @@ def unregister_model(self, app_label, model_name):
             del self.app_configs[app_label].models[model_name]
         except KeyError:
             pass
-        self.clear_cache()
```

Comment:
You need to call clear cache in `ProjectState.remove_model()` if you remove it here.